### PR TITLE
Remove IndicesNode and use Buffer directly

### PR DESCRIPTION
**Pull request by dj2**
This CL removes the IndicesNode and adds the Index buffer directly to
the Script object. The Executor is updated to pass all the set buffers
to the engine.

### CODE REVIEW

File: src/vkscript/executor.cc [hunk 1]
```diff
@@ -90,13 +90,8 @@ Result Executor::Execute(Engine* engine,
...
 
-  // Process Indices nodes
-  for (const auto& node : script->Nodes()) {
-    if (!node->IsIndices())
-      continue;
-
-    r = engine->SetBuffer(BufferType::kIndex, 0, Format(),
-                          node->AsIndices()->Indices());
+  for (const auto& buf : script->GetBuffers()) {
+    r = engine->SetBuffer(buf->GetBufferType(), 0, Format(), buf->GetData());
```
Comment (jaebaek) [inline src/vkscript/executor.cc]: I am confused. Why do we handle only the vertex data separately?
If we remove `IndicesNode`, I think we can do the same thing for `VertexDataNode`?

Comment (dj2) [reply src/vkscript/executor.cc]: That's the next CL which I'll put up for review when this lands.

Comment (dj2) [reply src/vkscript/executor.cc]: Vertex data has a more complicated Format than the indices DatumType. That has to be handled before vertex data can be converted.

Comment (jaebaek) [reply src/vkscript/executor.cc]: ok

**Pull request merged**